### ISSUE TITLE
Feature Request: Play local videos

### ISSUE THREAD

**Issue by yichengtan** [okld/streamlit-player#3]
While st.video works for local videos, it is very barebones in terms of features such as getting playback time and video parameters. Adding local video support for this would be great to utilize these features.

**Comment by Franky1** [comment]
I also was very confused, because the landing page here suggests that "Local Files" are supported, but i could not get it working.

```python
import streamlit as st
from streamlit_player import st_player

# official video player works:
st.video('video.mp4')

# this does not work:
st_player('video.mp4')
```

**Comment by ajaichemmanam** [comment]
Seting up a local server and using it as 

st_player("http://localhost:8520/archery.mp4",playing=True,playback_rate=1,progress_interval=1000, controls=True, loop=True)

works.

**Comment by Franky1** [comment]
@ajaichemmanam 
Yes, i know this is a possible solution, but it is way too much complicated and it won't work on streamlit cloud.
Furthermore you cannot use this solution with uploaded videos within the streamlit app.
Therefore this is not a viable solution...

**Comment by okld** [comment]
You can provide [custom `src` and `type`](https://github.com/CookPete/react-player#multiple-sources-and-tracks) as URL to ReactPlayer, which allows you to play a local video encoded in base64.

The following sample reads your video locally, encodes it in base64, and builds a valid data URI with the encoded data that can be played by ReactPlayer.

```python
from base64 import b64encode
from pathlib import Path
from streamlit_player import st_player

def local_video(path, mime="video/mp4"):
    data = b64encode(Path(path).read_bytes()).decode()
    return [{"type": mime, "src": f"data:{mime};base64,{data}"}]

st_player(local_video("path/to/video.mp4"))
```

**Comment by vivekdse** [comment]
> Seting up a local server and using it as
> 
> st_player("[http://localhost:8520/archery.mp4",playing=True,playback_rate=1,progress_interval=1000](http://localhost:8520/archery.mp4%22,playing=True,playback_rate=1,progress_interval=1000), controls=True, loop=True)
> 
> works.

Do you recommend any media server to do this job locally?

**Comment by vivekdse** [comment]
> You can provide [custom `src` and `type`](https://github.com/CookPete/react-player#multiple-sources-and-tracks) as URL to ReactPlayer, which allows you to play a local video encoded in base64.
> 
> The following sample reads your video locally, encodes it in base64, and builds a valid data URI with the encoded data that can be played by ReactPlayer.
> 
> ```python
> from base64 import b64encode
> from pathlib import Path
> from streamlit_player import st_player
> 
> def local_video(path, mime="video/mp4"):
>     data = b64encode(Path(path).read_bytes()).decode()
>     return [{"type": mime, "src": f"data:{mime};base64,{data}"}]
> 
> st_player(local_video("path/to/video.mp4"))
> ```

This works, which is the good news...but it is very slow with large media files.  Need to figure a way to stream the file rather than load it the whole way.

**Comment by Stinosko** [comment]
Using flask to serve the local files works for larger files but it is still not perfect.

```py
import os
from flask import Flask, send_file, make_response


APP = Flask(__name__)
MEDIA_PATH = 'path_to_media_folder'

@APP.route('/<path:vid_name>')
def serve_video(vid_name):
    print(vid_name)
    vid_path = os.path.join(MEDIA_PATH, vid_name)
    resp = make_response(send_file(vid_path, 'video/mp4'))
    resp.headers['Content-Disposition'] = 'inline'
    return resp


if __name__ == '__main__':
    APP.run()
```
In streamlit: `st_player("http://127.0.0.1:5000/"+  media["filepath"])`
